From: Simon McVittie <smcv@debian.org>
Date: Sun, 20 Feb 2022 21:23:25 +0000
Subject: cups: Use the same name-mangling as Debian 11's cups-browsed

The name-mangling used in GTK 3.24.25 is appropriate for
cups-browsed >= 1.28.11, which eliminates leading and trailing
underscores in the normalized name, but the cups-browsed version in
Debian 11 is older and does not do that. We need to match the printer
names generated by the installed cups-browsed, to avoid a printer
showing up twice under confusingly similar names.

Signed-off-by: Simon McVittie <smcv@debian.org>
Forwarded: not-needed, specific to Debian 11 software versions
---
 modules/printbackends/cups/gtkprintbackendcups.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/modules/printbackends/cups/gtkprintbackendcups.c b/modules/printbackends/cups/gtkprintbackendcups.c
index d5c00f9..eadcbad 100644
--- a/modules/printbackends/cups/gtkprintbackendcups.c
+++ b/modules/printbackends/cups/gtkprintbackendcups.c
@@ -3368,7 +3368,15 @@ avahi_service_resolver_cb (GObject      *source_object,
           printer_name_compressed_strv = g_new0 (gchar *, g_strv_length (printer_name_strv) + 1);
           for (i = 0, j = 0; printer_name_strv[i] != NULL; i++)
             {
-              if (printer_name_strv[i][0] != '\0')
+              /*
+               * Keep an empty segment at the beginning or end, if any.
+               * This emulates the way cups-browsed in Debian 11 created
+               * printer names (before 1.28.11), for example:
+               * "Some Printer (123456)" -> Some_Printer_123456_
+               * and hypothetically
+               * "(Really) Working Printer" -> _Really_Working_Printer
+               */
+              if (printer_name_strv[i][0] != '\0' || i == 0 || printer_name_strv[i + 1] == NULL)
                 {
                   printer_name_compressed_strv[j] = printer_name_strv[i];
                   j++;
